### PR TITLE
Autobuild: Minimize permissions in CI

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Reduces permissions in autobuild.yml to a bare minimum.

CHANGELOG: Internal: Reduced permissions in Autobuild for security hardening

**Context: Fixes an issue?**

Related: https://github.com/jamulussoftware/jamulus/issues/1737

**Does this change need documentation? What needs to be documented and how?**

**Status of this Pull Request**
Ready for review. Artifacts are building for a  release: https://github.com/ann0see/jamulus/actions/runs/3373043380 and seem to be ok. My repo has now enabled strict read only default permissons.

**What is missing until this pull request can be merged?**
Review

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: .github/workflows/autobuild.yml [hunk 1]
```diff
@@ -69,7 +70,8 @@ jobs:
       build_all_targets:            ${{ steps.decide-build-targets.outputs.build_all_targets }}
     env:
       release_changelog_path:       ./.github_release_changelog.md
-
+    permissions:
+        contents:                   write
```
Comment (ann0see) [inline .github/workflows/autobuild.yml:74]: Needed for release

File: .github/workflows/autobuild.yml [hunk 2]
```diff
@@ -179,6 +181,10 @@ jobs:
   release_assets:
     name:                           Build for ${{ matrix.config.config_name }}
     needs:                          create_release
+    permissions:
+      checks:                       write
+      contents:                     write
+      security-events:              write
```
Comment (ann0see) [inline .github/workflows/autobuild.yml:187]: Needed for artifact, release uploads and CodeQl

File: .github/workflows/autobuild.yml (hunk 1 above)
Comment (pljones) [reply .github/workflows/autobuild.yml:74]: Is it worth putting comments like this in the yaml?

Comment (ann0see) [reply .github/workflows/autobuild.yml:75]: Done.

**Review by hoffie — APPROVED** [review]
Thanks, looks good to me!

Did you try to build a test release on your repo with that logic in place? If not, it would be good to do so. :)

**Comment by ann0see** [comment]
Yes, see https://github.com/ann0see/jamulus/releases/tag/r0_0_0none4

I needed to set it as visible.

**Comment by ann0see** [comment]
@pljones Could you please review this?

**Review by pljones — APPROVED** [review]
Happy this grants no more than enough.  And happy it'll fail fast if it's not enough 😃.

**Comment by ann0see** [comment]
Now the other files also need updates :-).

**Pull request merged**